app/response/utils.py

```diff
@@ -222,12 +223,26 @@ def add_denial(request_id, reason_ids, email_content):
                 request_id,
                 es_update=False
             )
-        response = Determinations(
-            request_id,
-            RELEASE_AND_PUBLIC,
-            determination_type.DENIAL,
-            format_determination_reasons(reason_ids)
-        )
+        if not calendar.isbusday(datetime.utcnow()) or datetime.utcnow().date() < request.date_submitted.date():
+            # push the denial date to the next business day if it is a weekend/holiday
+            # or if it is before the date submitted
+            next_business_day = get_following_date(datetime.utcnow())
+            next_business_day = next_business_day.replace(hour=9, minute=00, second=00, microsecond=00)
+            next_business_day = local_to_utc(next_business_day, current_app.config['APP_TIMEZONE'])
```

Comment:
Can we make this a function in `app/lib/date_utils.py` since the code repeats twice (line 316-318)?